pkg/asset/machines/powervs/machines.go

```diff
@@ -0,0 +1,107 @@
+// Package powervs generates Machine objects for powerVS.
+package powervs
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+
+	machineapi "github.com/openshift/api/machine/v1beta1"
+	powervsprovider "github.com/openshift/cluster-api-provider-powervs/pkg/apis/powervsprovider/v1alpha1"
+	"github.com/openshift/installer/pkg/types"
+	"github.com/openshift/installer/pkg/types/powervs"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// Machines returns a list of machines for a machinepool.
+func Machines(clusterID string, config *types.InstallConfig, pool *types.MachinePool, role, userDataSecret string) ([]machineapi.Machine, error) {
+
```

Comment:
Do we need this here?
```suggestion
if configPlatform := config.Platform.Name(); configPlatform != powervs.Name {
	return nil, fmt.Errorf("non-powerVS configuration: %q", configPlatform)
}
```